torch/functional.py

```diff
@@ -1229,22 +1229,36 @@ def norm(input, p="fro", dim=None, keepdim=False, out=None, dtype=None):  # noqa
         p (int, float, inf, -inf, 'fro', 'nuc', optional): the order of norm. Default: ``'fro'``
             The following norms can be calculated:
 
-            =====  ============================  ==========================
-            ord    matrix norm                   vector norm
-            =====  ============================  ==========================
-            None   Frobenius norm                2-norm
-            'fro'  Frobenius norm                --
-            'nuc'  nuclear norm                  --
-            Other  as vec norm when dim is None  sum(abs(x)**ord)**(1./ord)
-            =====  ============================  ==========================
-
-        dim (int, 2-tuple of ints, 2-list of ints, optional): If it is an int,
-            vector norm will be calculated, if it is 2-tuple of ints, matrix norm
-            will be calculated. If the value is None, matrix norm will be calculated
-            when the input tensor only has two dimensions, vector norm will be
-            calculated when the input tensor only has one dimension. If the input
-            tensor has more than two dimensions, the vector norm will be applied to
-            last dimension.
+            ======  ==============  ==========================
+            ord     matrix norm     vector norm
+            ======  ==============  ==========================
+            'fro'   Frobenius norm  --
+            'nuc'   nuclear norm    --
+            Number  --              sum(abs(x)**ord)**(1./ord)
+            ======  ==============  ==========================
+
+            The vector norm can be calculated across any number of dimensions.
+            The corresponding dimensions of :attr:`input` are flattened into
+            one dimension, and the norm is calculated on the flattened
+            dimension.
+
+            Frobenius norm can be calculated across any number of dimensions,
```

Comment:
Wait, so `torch.norm(t, p='fro', ...)` and `torch.norm(t, p=2, ...)` are equivalent for all tensors `t` and values of the ellipses? That's wild.

If that's the case, should this just say that 'fro' is equivalent to p=2 and is a vector norm like it? That would also avoid this lengthy paragraph (although we could keep the Note as a proper Note::).

Reply:
I believe so, yes. It kind of defeats the purpose of having a separate `p` value for it. As long as `p='fro'` is not used with `len(dim) >= 3`, it should produce the same result as `p=2`, based on the formula for Frobenius norm (https://en.wikipedia.org/wiki/Matrix_norm#Frobenius_norm) and the limited set of cases I tried out. Although, we probably should have a unit test that shows this exhaustively, just to make sure there's not some special case I'm not considering.

Since `p='fro'` and `p=2` will act differently with respect to error conditions, we still need to point that out in the docs.

As a side note, Frobenius norm could have been implemented without a separate `at::frobenius_norm` function. So maybe an improvement going forward would be to remove that function.

Reply:
That makes sense. Are you happy with the current note, then, or do you update to say that 'fro' and p=2 are the same EXCEPT for some error checking that fro will do (when given a list of 3+ dims). And then maybe a note:: about how torch.linalg.norm handles the frobenius norm properly?

Reply:
Sure, I'll make those edits. Also I just wrote a test comparing fro and 2-norm (haven't gotten to run it yet though), I feel like it would be good to include the test in this PR just to back up my claim.

Reply:
Sounds great.

Reply:
Alright, I've made those updates and added the test. I think we should be good to go now